tests/unit/test_x509.cpp

```diff
@@ -568,3 +568,11 @@ TEST_F(X509Test, testGetSerialNumber)
     EXPECT_EQ("-42", negativeSerialCert.getSerialNumberDecimal());
     EXPECT_EQ(std::vector<uint8_t>{0x2a}, negativeSerialCert.getSerialNumberBinary());
 }
+
+TEST_F(X509Test, testCACheck)
+{
+    EXPECT_EQ(false, _root1_expired->isCA())
+        << "X509Certificate::isCA() should have returned false";
```

Comment:
This assert description repeats what is in the code and what will be reported by the test framework. It does not give you a better understanding of **why** the test is failing. I would you something like "Client Certificate cannot be CA" or, if your test-cases are small enough, remove messages at all.